### PR TITLE
impl(otel): batch TimeSeries

### LINKED ISSUE #14128: GCM exporter needs to batch requests
Cloud Monitoring enforces a limit of 200 TimeSeries / CreateTimeSeriesRequest. We need to enforce that in our exporter.

Note that we can fire these requests consecutively. The GCM minimum update period of 5 seconds applies per TimeSeries, and the TimeSeries in an `Export()` are unique.

### PR DESCRIPTION
**Pull request by dbolduc**
Fixes #14128 

Break up the export into multiple requests, each with no more than 200 TimeSeries in them. If we do not do this, we may drop all of our metrics on the floor if the cardinality is sufficiently high.

Generalize the existing tests to exercise the batching logic.

The error handling logic really is not important. I do not think anyone consumes the error. Arguably the tests are too strict.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14133)

### CODE REVIEW

File: google/cloud/opentelemetry/internal/time_series.cc [hunk 1]
```diff
@@ -218,16 +218,22 @@ std::vector<google::monitoring::v3::TimeSeries> ToTimeSeries(
...
+std::vector<google::monitoring::v3::CreateTimeSeriesRequest> ToRequests(
     std::string const& project, google::api::MonitoredResource const& mr_proto,
     std::vector<google::monitoring::v3::TimeSeries> tss) {
-  google::monitoring::v3::CreateTimeSeriesRequest request;
-  request.set_name(project);
+  std::vector<google::monitoring::v3::CreateTimeSeriesRequest> requests;
   for (auto& ts : tss) {
+    if (requests.empty() ||
+        requests.back().time_series().size() == kMaxTimeSeriesPerRequest) {
+      // If the current request is full (i.e. it has 200 TimeSeries), create a
```
Comment (devbww) [inline google/cloud/opentelemetry/internal/time_series.cc:228]: Maybe it's best to omit the parenthetical part?

File: google/cloud/opentelemetry/internal/time_series.h [hunk 2]
```diff
@@ -72,8 +77,13 @@ std::vector<google::monitoring::v3::TimeSeries> ToTimeSeries(
 
 /**
  * Convert from OpenTelemetry metrics to Cloud Monitoring protos.
+ *
+ * We return a vector of requests, because Cloud Monitoring accepts at most 200
```
Comment (devbww) [inline google/cloud/opentelemetry/internal/time_series.h:81]: Leave the literal "200" to just the constant declaration?

File: google/cloud/opentelemetry/monitoring_exporter.cc [hunk 3]
```diff
@@ -69,22 +69,35 @@ class MonitoringExporter final
  private:
   opentelemetry::sdk::common::ExportResult ExportImpl(
       opentelemetry::sdk::metrics::ResourceMetrics const& data) {
+    auto result = opentelemetry::sdk::common::ExportResult::kSuccess;
+
     auto tss = ToTimeSeries(data, prefix_);
     if (tss.empty()) {
       GCP_LOG(INFO) << "Cloud Monitoring Export skipped. No data.";
-      return opentelemetry::sdk::common::ExportResult::kSuccess;
+      // Return early to save the littlest bit of processing.
```
Comment (devbww) [inline google/cloud/opentelemetry/monitoring_exporter.cc:77]: :-)

**Review by devbww — APPROVED** [review]

File: google/cloud/opentelemetry/internal/time_series.h (hunk 2 above)
Comment (dbolduc) [reply google/cloud/opentelemetry/internal/time_series.h]: Yeah you are right. `200` should really only show up in one place. Done.

**Pull request merged**